packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts

```diff
@@ -124,12 +124,12 @@ export default mixins(
 
       return this.$createElement('button', setColor(color, {
         staticClass: 'v-btn',
-        class: this.genButtonClasses(isAllowed, isFloating, isSelected, isCurrent),
+        class: this.genButtonClasses(isAllowed && !isOtherMonth, isFloating, isSelected, isCurrent),
         attrs: {
           type: 'button',
         },
         domProps: {
-          disabled: this.disabled || !isAllowed,
+          disabled: this.disabled || !isAllowed || isOtherMonth,
```

Comment:
idk if disabled is right, the native ones are greyed out but still clickable: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date